### PR TITLE
th-213: Business Journey - FE: Order List

### PR DESCRIPTION
**Pull request by TetianaDoroshko**
fetch all business orders and display them using Order List Card for business Component
 add google map component to the right, with markers where the final destination of orders are
 While hovering on the order - on map should be displayed start and end points, with a proposed route between.
 
 Note: Changed Map servise - MapService conscructor  should accept either the center or the bounds (if the center is unknown)
 
image

### CODE REVIEW

File: frontend/.env.example [hunk 1]
```diff
@@ -8,4 +8,4 @@ VITE_APP_PROXY_SERVER_URL=http://localhost:3001
 
 # API
 VITE_APP_GOOGLE_MAPS_API_KEY=your_google_maps_api_key
-DRIVER_GEOLOCATION_UPDATE_INTERVAL=interval_in_ms
+DRIVER_GEOLOCATION_UPDATE_INTERVAL=5000
```
Comment (denred) [inline frontend/.env.example]: Why do you define 5s in the example?

File: frontend/src/libs/components/map/map-inner-component/map-inner-component.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}: Properties) => {
+  const mapReference = useRef<HTMLDivElement>(null);
+  const mapService = useRef<MapService | null>(null);
+  const mapClasses = getValidClassNames(styles.map, className);
+
+  const showRoute = useCallback(async (): Promise<void> => {
+    if (!shownRoute) {
+      return;
+    }
+    mapService.current && (await mapService.current.addRoute(shownRoute));
```
Comment (denred) [inline frontend/src/libs/components/map/map-inner-component/map-inner-component.tsx]: return shownRoute
  ? mapService.current && (await mapService.current.addRoute(shownRoute))
  : undefined;  
It seems better to use a ternary operator.

File: frontend/src/libs/components/map/map.tsx [hunk 3]
```diff
@@ -34,20 +41,38 @@ const Map: React.FC<Properties> = ({
...
-      if (center && destination) {
-        mapService.current.addMarker(destination);
+      return;
+    }
+
+    mapService.current = new MapService({
+      mapElement: mapReference.current,
+      center,
+      zoom,
+    });
```
Comment (denred) [inline frontend/src/libs/components/map/map.tsx]: Why not create the object once and change only the options?

File: frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { OrderStatus } from 'shared/build';
+import { type OrderStatusValues } from 'shared/build/packages/orders/libs/types/order-status-values.type';
+
+import { type Color } from '~/libs/types/color.type';
```
Comment (denred) [inline frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts]: Check the imports, it seems, they should be from the packages and with .js.

File: frontend/src/libs/components/order-list-card/order-list-card-business.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
+import { type OrderResponseDto } from 'shared/build/index.js';
+
+import { IconName } from '~/libs/enums/icon-name.enum';
```
Comment (denred) [inline frontend/src/libs/components/order-list-card/order-list-card-business.tsx]: check imports

File: frontend/src/libs/components/order-list/order-list.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { type OrderResponseDto } from 'shared/build';
```
Comment (denred) [inline frontend/src/libs/components/order-list/order-list.tsx]: check imports

File: frontend/src/libs/packages/map/map.package.ts [hunk 7]
```diff
@@ -15,29 +15,56 @@ type Constructor = {
...
+  private infoWindow: google.maps.InfoWindow;
+
+  private geoCoder: google.maps.Geocoder;
+
+  public constructor({ mapElement, center, zoom, bounds }: Constructor) {
     this.directionsService = new google.maps.DirectionsService();
     this.directionsRenderer = new google.maps.DirectionsRenderer({
       suppressMarkers: true,
     });
+    this.infoWindow = new google.maps.InfoWindow({ maxWidth: 250 });
```
Comment (denred) [inline frontend/src/libs/packages/map/map.package.ts]: There should probably be a constant.

**Review by denred — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/order-list-card/order-list-card-business.tsx (hunk 5 above)
Comment (Some14u) [reply frontend/src/libs/components/order-list-card/order-list-card-business.tsx]: should be with .js
also, we use direct shared import only inside /libs/types/types.js, /libs/enums/enums.js e.t.c., so basically your package should always have some of them, which would then be used to import to any of you package modules.

File: frontend/src/libs/components/order-list-card/order-list-card-business.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  select: ({
+    startPoint,
+    endPoint,
+  }: {
+    startPoint: google.maps.LatLngLiteral;
+    endPoint: google.maps.LatLngLiteral;
+  }) => void;
+};
+
+const OrderListCardBusiness: React.FC<Properties> = ({
```
Comment (Some14u) [inline frontend/src/libs/components/order-list-card/order-list-card-business.tsx]: The name of the module should be the same as the name of the component inside. Either rename the module to order-list-card-business.tsx (along with the folder) or the component to OrderListCard

File: frontend/src/libs/packages/map/map.package.ts [hunk 9]
```diff
@@ -154,12 +181,12 @@ class MapService implements IMapService {
     position: google.maps.LatLngLiteral,
     isOrigin = false,
     angle = 0,
-  ): void {
+  ): google.maps.Marker {
     this.throwIfMapNotInitialized();
 
     const rotatedIconUrl = rotateImg(truckImg, angle);
```
Comment (Some14u) [inline frontend/src/libs/packages/map/map.package.ts]: can you extract the logic of icon creation to helper?
it can be something like 
```
const createRotatedTruckIcon = (
  isOrigin: boolean,
  angle: number, 
  scale: number = DEFAULT_TRUCK_SCALE
) => {
   if (!isOrigin) {
      return truckImg;
   }
   const url = rotateImg(truckImg, angle);
   const anchor = new google.maps.Point(
              TRUCK_IMG_ANCHOR_X,
              TRUCK_IMG_ANCHOR_Y,
            );
   const size = new google.maps.Size(TRUCK_IMG_WIDTH, TRUCK_IMG_HEIGHT);
   return { url, anchor, size, scale }
}
```

File: frontend/src/pages/dashboard/components/orders/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,37 @@
+@import "src/assets/css/vars.scss";
+
+.orders {
+  display: flex;
+  flex-direction: column;
+  height: max-content;
+  min-height: 100%;
+  padding: 20px;
+  background-color: #ffffff;
```
Comment (Some14u) [inline frontend/src/pages/dashboard/components/orders/styles.module.scss]:
```suggestion
  background-color: $white;
```

File: frontend/src/pages/dashboard/components/orders/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  }
+}
+
+.mapWrapper {
+  position: sticky;
+  top: 0;
+  height: calc(100vh - 160px);
+  overflow: hidden;
+  border-radius: 20px;
+  box-shadow: 0 4px 4px #00000021;
```
Comment (Some14u) [inline frontend/src/pages/dashboard/components/orders/styles.module.scss]: is it okay to use `$black-light-transparent` instead of `#00000021` here?

File: frontend/src/pages/dashboard/components/orders/orders.tsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  }, [orders]);
+
+  return (
+    <div className={styles.orders}>
+      <LoadScript
+        googleMapsApiKey={config.ENV.API.GOOGLE_MAPS_API_KEY}
+        libraries={libraries}
+      >
+        <div className={styles.orderlistArea}>
+          <div>Filter</div>
```
Comment (Some14u) [inline frontend/src/pages/dashboard/components/orders/orders.tsx]: No filter yet?

File: frontend/src/pages/dashboard/components/orders/orders.tsx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+          <OrderList orders={orders} select={setShownRoute} />
+        </div>
+        <div className={styles.mapArea}>
+          <div className={styles.mapWrapper}>
+            <Map
+              zoom={10}
```
Comment (Some14u) [inline frontend/src/pages/dashboard/components/orders/orders.tsx]: Is this '10' temporary?

**Review by Some14u — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/order-list/order-list.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+const OrderList: React.FC<Properties> = ({ orders, select }: Properties) => {
+  return (
+    <ul className={styles.orderList}>
+      {orders.map((order) => (
+        <li key={order.id} className={styles.orderItem}>
+          <OrderListCardBusiness
+            order={order}
+            select={select}
+          ></OrderListCardBusiness>
```
Comment (AlexVlaso) [inline frontend/src/libs/components/order-list/order-list.tsx]:
```suggestion
          <OrderListCardBusiness
            order={order}
            select={select}
          />
```

File: frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+      return { name: 'Canceled', color: 'red' };
+    }
+    case OrderStatus.DONE: {
+      return { name: 'Done', color: 'grey-dark' };
+    }
+    case OrderStatus.PICKING_UP: {
+      return { name: 'Arrived', color: 'green' };
+    }
+    default: {
+      return { name: 'Pending', color: 'grey-light' };
```
Comment (AlexVlaso) [inline frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts]: I think we should add additional var to the palette. For example for canceled badges you used the color "#ee2a64" but in our layout  we have "#ee2a6426" the same for green

File: frontend/.env.example (hunk 1 above)
Comment (TetianaDoroshko) [reply frontend/.env.example]: it was accidentally

File: frontend/src/libs/components/map/map-inner-component/map-inner-component.tsx (hunk 2 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/components/map/map-inner-component/map-inner-component.tsx]: Map inner component is deprecated now, I've remove it.

File: frontend/src/libs/components/map/map.tsx (hunk 3 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/components/map/map.tsx]: Because Map service can be initialized whether with center coordinates or with bounds ( if there are a few marcers and the center is unknown).

File: frontend/src/libs/components/map/map.tsx [hunk 16]
```diff
@@ -5,14 +5,18 @@ import { actions as orderActions } from '~/slices/orders/order.js';
 
 import styles from './styles.module.scss';
 
-const DEFAULT_ZOOM = 16;
+const DEFAULT_ZOOM = 10;
```
Comment (Some14u) [inline frontend/src/libs/components/map/map.tsx]: Move this to libs/consts

**Review by Some14u — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts (hunk 15 above)
Comment (Some14u) [reply frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts]: We can use $red as a base color, and derive a required transparent variation of it by using [sass color functions](https://sass-lang.com/documentation/modules/color/). There is a working example in frontend\src\assets\css\plugins\balloon-css.plugin.scss

Comment (Some14u) [reply frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts]: Besides that, @mV22110702 found a way to import sass variables into js.
The example of this you can find in his frontend\src\libs\components\chosen-file-preview\chosen-file-preview.tsx
look at definition zoomOutSpeed constant (the value for it is in scss file)
I  believe, this can be used to get the colors from our global sass variables and apply them to badges.
Another option is to overhaul badges so they would use colors from the vars.sass in combination with specific classes.

File: frontend/src/libs/components/order-list-card/order-list-card-business.tsx (hunk 8 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/components/order-list-card/order-list-card-business.tsx]: I have done so, because we will also have a order list car for customers. And I supposed to hold it inside order-list-card directory. But ok, I've changed.

File: frontend/src/pages/dashboard/components/orders/orders.tsx (hunk 12 above)
Comment (TetianaDoroshko) [reply frontend/src/pages/dashboard/components/orders/orders.tsx]: in another branch

File: frontend/src/pages/dashboard/components/orders/orders.tsx (hunk 13 above)
Comment (TetianaDoroshko) [reply frontend/src/pages/dashboard/components/orders/orders.tsx]: no. the Map component requires that prop. If it is not passed, Map will apply a default zoom.

File: frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts (hunk 15 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/components/order-list-card/libs/helpers/status-converter/status-converter.helper.ts]: We can use the palette from the Badge component only. Or extend it. I suggest use the existed palette.

File: frontend/src/assets/css/vars.scss [hunk 17]
```diff
@@ -1,6 +1,6 @@
 $black-light: #1a202c;
 $black: #1d1d1d;
-$black-light-transparent: rgb(0 0 0 / 5%);
+$black-light-transparent: hwb(0deg 0% 100% / 5%);
```
Comment (Some14u) [inline frontend/src/assets/css/vars.scss]: why did you change this?

File: frontend/src/libs/packages/map/map.package.ts [hunk 18]
```diff
@@ -1,43 +1,66 @@
...
+    mapElement,
+    center,
+    zoom,
+    bounds,
+  }: {
+    mapElement: HTMLDivElement;
+    center?: google.maps.LatLngLiteral;
+    zoom?: number;
+    bounds?: google.maps.LatLngBounds;
+  }): void {
```
Comment (Some14u) [inline frontend/src/libs/packages/map/map.package.ts]: You already have this type, it's called Constructor. You can reuse it (don't forget to rename, Constructor will not suit anymore because of usage in multiple places).

**Review by Some14u — CHANGES_REQUESTED** [review]

File: frontend/src/assets/css/vars.scss (hunk 17 above)
Comment (TetianaDoroshko) [reply frontend/src/assets/css/vars.scss]: a don't know how have I made it O_o

**Review by Some14u — APPROVED** [review]

File: frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+
+const createIcon = (isOrigin: boolean, angle: number): string | Icon => {
+  const rotatedIconUrl = rotateImg(truckImg, angle);
+
+  return isOrigin
+    ? {
+        url: rotatedIconUrl,
+        anchor: new google.maps.Point(TRUCK_IMG_ANCHOR_X, TRUCK_IMG_ANCHOR_Y),
+        size: new google.maps.Size(TRUCK_IMG_WIDTH, TRUCK_IMG_HEIGHT),
+        scale: 1,
```
Comment (denred) [inline frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts]: Is it probably const too?

**Review by denred — CHANGES_REQUESTED** [review]

File: frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts (hunk 19 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts]: it is constants

**Review by MaxHopaytsa — APPROVED** [review]

File: frontend/src/slices/orders/order.slice.ts [hunk 20]
```diff
@@ -43,6 +43,17 @@ const { reducer, actions, name } = createSlice({
       })
       .addCase(calculateOrderPrice.rejected, (state) => {
         state.dataStatus = DataStatus.REJECTED;
+      })
+      .addCase(getOrders.pending, (state) => {
+        state.orders = [];
+        state.dataStatus = DataStatus.FULFILLED;
```
Comment (kalinkaaaa14) [inline frontend/src/slices/orders/order.slice.ts]:
```suggestion
        state.dataStatus = DataStatus.PENDING;
```

File: frontend/src/slices/orders/order.slice.ts [hunk 21]
```diff
@@ -43,6 +43,17 @@ const { reducer, actions, name } = createSlice({
       })
       .addCase(calculateOrderPrice.rejected, (state) => {
         state.dataStatus = DataStatus.REJECTED;
+      })
+      .addCase(getOrders.pending, (state) => {
+        state.orders = [];
```
Comment (kalinkaaaa14) [inline frontend/src/slices/orders/order.slice.ts]: redundant

File: frontend/src/pages/dashboard/components/orders/orders.tsx (hunk 13 above)
Comment (kalinkaaaa14) [inline frontend/src/pages/dashboard/components/orders/orders.tsx]: don't use magic num

File: frontend/src/packages/orders/orders-api.ts [hunk 22]
```diff
@@ -22,6 +22,19 @@ class OrdersApi extends HttpApi {
     super({ path: ApiPath.ORDERS, baseUrl, http, storage });
   }
 
+  public async getOrders(): Promise<OrderResponseDto[]> {
```
Comment (kalinkaaaa14) [inline frontend/src/packages/orders/orders-api.ts]:
```suggestion
  public async getBusinessOrders(): Promise<OrderResponseDto[]> {
```

File: frontend/src/libs/packages/map/map.package.ts [hunk 23]
```diff
@@ -155,26 +168,60 @@ class MapService implements IMapService {
...
+  }
+
+  public async getAddress(place: google.maps.LatLngLiteral): Promise<string> {
+    this.throwIfMapNotInitialized();
+
+    const result: google.maps.GeocoderResponse = await this.geoCoder.geocode({
+      location: place,
+    });
+
+    return result.results[2].formatted_address;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/map/map.package.ts]: what means result.results[2] ?

File: frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts (hunk 19 above)
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts]: avoid magic numb

File: frontend/src/libs/components/order-list/order-list.tsx [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+
+type Properties = {
+  orders: OrderResponseDto[];
+  select: ({
+    startPoint,
+    endPoint,
+  }: {
+    startPoint: google.maps.LatLngLiteral;
+    endPoint: google.maps.LatLngLiteral;
+  }) => void;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list/order-list.tsx]: create a type for it

File: frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+
+type Properties = {
+  order: OrderResponseDto;
+  select: ({
+    startPoint,
+    endPoint,
+  }: {
+    startPoint: google.maps.LatLngLiteral;
+    endPoint: google.maps.LatLngLiteral;
+  }) => void;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx]: reuse created type here

File: frontend/src/libs/components/order-list-card-business/libs/helpers/status-converter/status-converter.helper.ts [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+      return { name: 'Done', color: 'grey-dark' };
+    }
+    case OrderStatus.PICKING_UP: {
+      return { name: 'Arrived', color: 'green' };
+    }
+    default: {
+      return { name: 'Pending', color: 'grey-light' };
+    }
+  }
+};
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/libs/helpers/status-converter/status-converter.helper.ts]: it should be map, not switch case

**Review by kalinkaaaa14 — CHANGES_REQUESTED** [review]

File: frontend/src/libs/packages/map/map.package.ts (hunk 23 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/packages/map/map.package.ts]: GeocoderResult looks like an array of address objects with different formats
image

File: frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts (hunk 19 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts:20]: I've just taken out it in a separate helper

Comment (Some14u) [reply frontend/src/libs/packages/map/libs/helpers/create-icon.helper.ts]: I'd suggest you to move this scale to parameters of createIcon, because it might be used in the future to control the scale of the icon dynamically.

**Review by Some14u — APPROVED** [review]

**Review by denred — APPROVED** [review]

**Review by MaxHopaytsa — APPROVED** [review]

File: frontend/src/libs/enums/img-path.enum.ts [hunk 27]
```diff
@@ -0,0 +1,7 @@
+import avatarDefault from '~/assets/img/avatar-default.png';
+
+const ImgPath = {
+  AVAVTAR_DEFAULT: avatarDefault,
```
Comment (kalinkaaaa14) [inline frontend/src/libs/enums/img-path.enum.ts]:
```suggestion
  AVATAR_DEFAULT: avatarDefault,
```

Comment (kalinkaaaa14) [reply frontend/src/libs/enums/img-path.enum.ts]: and actually do we need an enum for it?

File: frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+const statusMapper = new Map<OrderStatusValues, { name: string; color: Color }>(
+  [
+    [OrderStatus.PENDING, { name: 'Pending', color: 'grey-light' }],
+    [OrderStatus.CONFIRMED, { name: 'On the way', color: 'blue-light' }],
+    [OrderStatus.CANCELED, { name: 'Canceled', color: 'red' }],
+    [OrderStatus.DONE, { name: 'Done', color: 'grey-dark' }],
+    [OrderStatus.PICKING_UP, { name: 'Arrived', color: 'green' }],
+  ],
+);
+
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts]: you should create a map, where key will be OrderStatus, the value will be object with readable value + needed color
const mapOrderStatusToReadable

File: frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts:1]: rename file

File: frontend/src/libs/packages/map/map.package.ts [hunk 29]
```diff
@@ -155,26 +169,56 @@ class MapService implements IMapService {
...
+  public async getAddress(place: google.maps.LatLngLiteral): Promise<string> {
+    this.throwIfMapNotInitialized();
+
+    const result: google.maps.GeocoderResponse = await this.geoCoder.geocode({
+      location: place,
     });
+
+    const [address] = result.results;
+
+    return address.formatted_address;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/packages/map/map.package.ts]:
```suggestion
    const [address] = result.results;
    const { formatted_address: formattedAddress} = address;
    
    return formattedAddress;
```

File: frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts [hunk 30]
```diff
@@ -0,0 +1,15 @@
+import { OrderStatus } from '~/libs/enums/enums.js';
+import { type Color } from '~/libs/types/color.type';
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts]:
```suggestion
import { type Color } from '~/libs/types/color.type.js';
```

File: frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts [hunk 31]
```diff
@@ -0,0 +1,15 @@
...
+import { type OrderStatusValues } from '~/libs/types/types.js';
+
+const statusMapper = new Map<OrderStatusValues, { name: string; color: Color }>(
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts]: Sounds like a function. Let's rename this one to something like `orderStatusToBadgeProps`?

File: frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+}: Properties) => {
+  const {
+    id,
+    status,
+    startPoint,
+    endPoint,
+    shift: { driver, truck },
+  } = order;
+
+  const selectCard = useCallback(
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx]: handleSelectCard

File: frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+import { type PlaceLatLng } from '~/libs/packages/map/libs/types/types.js';
+import { type OrderResponseDto } from '~/libs/types/types.js';
+
+import { Badge, Icon } from '../components.js';
+import { statusMapper } from './libs/mapper/status-mapper/status.mapper.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  order: OrderResponseDto;
+  select: ({ startPoint, endPoint }: PlaceLatLng) => void;
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx]: onSelect

File: frontend/src/libs/hooks/use-app-map/use-app-map.hook.ts [hunk 34]
```diff
@@ -36,11 +43,32 @@ const useAppMap = ({
...
-        center: center ?? DEFAULT_CENTER,
-        zoom,
-      });
+      if (!points) {
+        mapService.current = new MapConnector().getMapService({
+          mapElement: mapReference.current,
+          center: center ?? DEFAULT_CENTER,
+          zoom,
+        });
+      } else if (points.length === 1) {
```
Comment (PavloSukhinin) [inline frontend/src/libs/hooks/use-app-map/use-app-map.hook.ts]: We should show our intentions and try to answer a question: why exactly one point? Let's introduce a named helper for this.

File: frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+      </div>
+      <div className={styles.content}>
+        <img
+          src={ImgPath.AVATAR_DEFAULT}
+          alt={driver.firstName}
+          className={styles.avatar}
+        />
+        <div className={styles.driver}>
+          <p className={getValidClassNames('textMdBold')}>
+            {driver.firstName} {driver.lastName}
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/order-list-card-business.tsx]: we are having a helper for this in shared - getFullName

File: frontend/src/libs/components/order-list-card-business/libs/map/status-name.map.ts [hunk 36]
```diff
@@ -0,0 +1,13 @@
+import { OrderStatus } from '~/libs/enums/enums.js';
+import { type Color } from '~/libs/types/color.type';
+import { type OrderStatusValues } from '~/libs/types/types.js';
+
+const statusName: Record<OrderStatusValues, { name: string; color: Color }> = {
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list-card-business/libs/map/status-name.map.ts]:
```suggestion
type ReadableStatusWithColor = {
      name: string;
      color: Color;
}
const mapOrderStatusToReadable: Record<OrderStatusValues, ReadableStatusWithColor> = {
```

Comment (kalinkaaaa14) [reply frontend/src/libs/components/order-list-card-business/libs/map/status-name.map.ts]: + don't forget about file renaming

File: frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts (hunk 31 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/components/order-list-card-business/libs/mapper/status-mapper/status.mapper.ts]: I've replace it to plain object

File: frontend/src/libs/hooks/use-app-map/use-app-map.hook.ts (hunk 34 above)
Comment (TetianaDoroshko) [reply frontend/src/libs/hooks/use-app-map/use-app-map.hook.ts]: It's because if there is only one point in the list we can define this point as a center and init map with center and not get bounds.

**Review by PavloSukhinin — DISMISSED** [review]
LGTM! Just waiting for solved conflicts.

File: frontend/src/libs/components/order-list/order-list.tsx [hunk 37]
```diff
@@ -0,0 +1,24 @@
+import { type PlaceLatLng } from '~/libs/packages/map/libs/types/types.js';
+import { type OrderResponseDto } from '~/libs/types/types.js';
+
+import { OrderListCardBusiness } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  orders: OrderResponseDto[];
+  select: ({ startPoint, endPoint }: PlaceLatLng) => void;
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/order-list/order-list.tsx]:
```suggestion
  onSelect: ({ startPoint, endPoint }: PlaceLatLng) => void;
```

Comment (TetianaDoroshko) [reply frontend/src/libs/components/order-list/order-list.tsx]: I definitely changed it. I have no idea why it hasn't changed. Maybe autoresolve conflicts. Redid.

**Review by PavloSukhinin — APPROVED** [review]
